charms/knative-eventing/src/charm.py

```diff
@@ -4,101 +4,85 @@
-        # Get a reference the container attribute on the PebbleReadyEvent
-        container = event.workload
-        # Define an initial Pebble layer configuration
-        pebble_layer = {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
-            "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {"thing": self.model.config["thing"]},
-                }
-            },
-        }
-        # Add initial Pebble config layer using the Pebble API
-        container.add_layer("httpbin", pebble_layer, combine=True)
-        # Autostart any services that were defined with startup: enabled
-        container.autostart()
-        # Learn more about statuses in the SDK docs:
-        # https://juju.is/docs/sdk/constructs#heading--statuses
-        self.unit.status = ActiveStatus()
-
-    def _on_config_changed(self, _):
-        """Just an example to show how to deal with changed configuration.
-
-        TEMPLATE-TODO: change this example to suit your needs.
-        If you don't need to handle config, you can remove this method,
-        the hook created in __init__.py for it, the corresponding test,
-        and the config.py file.
-
-        Learn more about config at https://juju.is/docs/sdk/config
-        """
-        current = self.config["thing"]
-        if current not in self._stored.things:
-            logger.debug("found a new thing: %r", current)
-            self._stored.things.append(current)
-
-    def _on_fortune_action(self, event):
-        """Just an example to show how to receive actions.
-
-        TEMPLATE-TODO: change this example to suit your needs.
-        If you don't need to handle actions, you can remove this method,
-        the hook created in __init__.py for it, the corresponding test,
-        and the actions.py file.
-
-        Learn more about actions at https://juju.is/docs/sdk/actions
-        """
-        fail = event.params["fail"]
-        if fail:
-            event.fail(fail)
+        self.framework.observe(self.on.remove, self._on_remove)
+
+    def _on_install(self, _):
+        try:
+            self.resource_handler.apply()
+        except (ApiError, ErrorWithStatus) as e:
+            if isinstance(e, ApiError):
+                logger.debug(traceback.format_exc())
+                logger.info(
+                    f"Applying resources failed with ApiError status code {str(e.status.code)}"
+                )
+                self.unit.status = BlockedStatus(f"ApiError: {str(e.status.code)}")
+            elif isinstance(e, ReconcileError):
+                logger.info(f"Applying resources failed with message {str(e.msg)}")
+                self.unit.status = BlockedStatus(f"ErrorWithStatus: {str(e.msg)}")
```

Comment:
Two comments:
1) based on your feedback, I removed the `ReconcileError` entirely (because it wasn't being raised for a reconcile anyway, and the one cases it was being raised didn't feel like something that needed its own exception).  So this'll now be a `ErrorWithStatus`
2) I'd trust the upstream function raising the error, and change this to:
```python
        except (ApiError, ErrorWithStatus) as e:
            # This is a case where we have an opinion, so we handle it specifically
            if isinstance(e, ApiError):
                logger.debug(traceback.format_exc())
                logger.info(
                    f"Applying resources failed with ApiError status code {str(e.status.code)}"
                )
                self.unit.status = BlockedStatus(f"ApiError: {str(e.status.code)}")
            else:
                logger.info(str(e.msg))  # Maybe you could add something here about context from the parent charm, but I dont think you need to
                self.unit.status = e.status  # Trust the raiser to assign a status that is right for this situation, unless we know a case where we shouldnt
```